File: 2nd year/MPP/project-mpp/csharp-project/model/Referee.cs
namespace model
{
    public class Referee : Entity<long>
    {
        public Game Game { get; set; }
        public string Name { get; set; }
        public string Username { get; set; }
        public string Password { get; set; }

       /* public Referee(Game game, string name, string username, string password) : this(0L, game, name, username, password)
        {

        }

        public Referee(long id, Game game, string name, string username, string password)
        {
            Id = id;
            Game = game;
            Name = name;
            Username = username;
            Password = password;
        } */

        public override string ToString()
        {
            return "{\"id\": " + Id +
                ", \"game\": " + Game.ToString() +
                ", \"name\": \"" + Name + "\"" +
                ", \"username\": \"" + Username + "\"" +
                ", \"password\": \"" + Password + "\"" + "}";
        }
    }
}

File: 2nd year/MPP/project-mpp/java-project/TriathlonModel/src/main/java/triathlon/model/Referee.java
package triathlon.model;

import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import javax.persistence.Entity;
import java.io.Serializable;

@Entity
@Table(name = "Referee")
public class Referee extends triathlon.model.Entity<Long> implements Serializable {
    private Game game;
    private String name;
    private String username;
    private String password;

    public Referee(Long id, Game game, String name, String username, String password) {
        setId(id);
        this.game = game;
        this.name = name;
        this.username = username;
        this.password = password;
    }

    public Referee(Game game, String name, String username, String password) {
        this.game = game;
        this.name = name;
        this.username = username;
        this.password = password;
    }

    public Referee() {}

    @Override
    public void setId(Long id) {
        super.setId(id);
    }

    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    @GenericGenerator(name="identity", strategy = "identity")
    @Override
    public Long getId() {
        return super.getId();
    }

    @ManyToOne
    @JoinColumn(name = "gameId", referencedColumnName = "id")
    public Game getGame() {
        return game;
    }

    public void setGame(Game game) {
        this.game = game;
    }

    @Column(name = "name")
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Column(name = "username")
    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    @Column(name = "password")
    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    @Override
    public String toString() {
        return "Referee{" +
                "id=" + getId() +
                ", game=" + game +
                ", name='" + name + '\'' +
                ", username='" + username + '\'' +
                ", password='" + password + '\'' +
                '}';
    }
}